daemon/metrics.go

```diff
@@ -40,3 +49,64 @@ func init() {
 	imageActions = ns.NewLabeledTimer("image_actions", "The number of seconds it takes to process each image action", "action")
 	metrics.Register(ns)
 }
+
+const metricsPluginType = "MetricsCollector"
+
+func (d *Daemon) cleanupMetricsPlugins() {
+	ls := d.PluginStore.GetAllManagedPluginsByCap(metricsPluginType)
+	var wg sync.WaitGroup
+	wg.Add(len(ls))
+
+	for _, p := range ls {
+		go func() {
+			defer wg.Done()
+			stopMetricsPlugin(p)
+		}()
+	}
+	wg.Wait()
+
+	if d.metricsPluginListener != nil {
+		d.metricsPluginListener.Close()
+	}
+}
+
+type metricsPlugin struct {
+	plugingetter.CompatPlugin
+}
+
+func (p metricsPlugin) sock() string {
+	return "metrics.sock"
+}
+
+func (p metricsPlugin) sockBase() string {
+	return filepath.Join(p.BasePath(), "run", "docker")
+}
+
+func startMetricsPlugin(p plugingetter.CompatPlugin) error {
```

Comment:
I think `StartMetrics` and `StopMetrics` are ok because this is called once per daemon per plugin.
We could do `StartCollection` and `StopCollection`, something else?

Agree log messages are not great here.